pallets/process-validation/src/lib.rs

```diff
@@ -90,38 +92,140 @@ pub mod pallet {
         ValueQuery,
     >;
 
+    #[pallet::storage]
+    #[pallet::getter(fn version_model)]
+    pub(super) type VersionModel<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::ProcessIdentifier, T::ProcessVersion, ValueQuery>;
+
     #[pallet::event]
+    // looking by the type, same type for multiple things - bnad idea
+    #[pallet::metadata(
+        ProcessIdentifier = "Process Id",
+        ProcessVersion = "Process Version",
+        bool = "Is New"
```

Comment:
Not sure if these can/should include spaces. You may want to look into metadata usages elsewhere in substrate to get an idea of what's idiomatic

Reply:
updated